boa_engine/src/object/shape/slot.rs

```diff
@@ -0,0 +1,86 @@
+use bitflags::bitflags;
+pub(crate) type SlotIndex = u32;
+
+bitflags! {
+    /// Attributes of a slot.
+    #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    pub struct SlotAttribute: u8 {
+        const WRITABLE = 0b0000_0001;
+        const ENUMERABLE = 0b0000_0010;
+        const CONFIGURABLE = 0b0000_0100;
+        const HAS_GET = 0b0000_1000;
+        const HAS_SET = 0b0001_0000;
+    }
+}
+
+impl SlotAttribute {
+    pub const fn is_accessor_descriptor(self) -> bool {
+        self.contains(Self::HAS_GET) || self.contains(Self::HAS_SET)
+    }
+    pub const fn is_data_descriptor(self) -> bool {
+        !self.is_accessor_descriptor()
+    }
+
+    pub const fn has_get(self) -> bool {
+        self.contains(Self::HAS_GET)
+    }
+    pub const fn has_set(self) -> bool {
+        self.contains(Self::HAS_SET)
+    }
+
+    /// Check if slot type width matches, this can only happens,
+    /// if they are both accessors, or both data properties.
+    pub const fn width_match(self, other: Self) -> bool {
+        self.is_accessor_descriptor() == other.is_accessor_descriptor()
+    }
+
+    /// Get the width of the slot.
+    pub fn width(self) -> u32 {
+        // accessor take 2 positions in the storage to accomodate for the `get` and `set` fields.
+        1 + u32::from(self.is_accessor_descriptor())
+    }
```

Comment:
Could you explain this a bit more, im not sure i fully understand. Accessor slots will jump the index by 2 from what I can see, whats the reason this needs to happen?

Reply:
slots are varying in length, accessors have width 2 (get and set ) while data properties have with 1 (JsValue data)